src/script/E2EIdentity/E2EIdentityEnrollment.ts

```diff
@@ -117,107 +112,68 @@ export class E2EIHandler extends TypedEventEmitter<Events> {
     return this;
   }
 
-  private async processEnrollmentUponExpiry() {
-    const hasCertificate = await hasActiveCertificate();
-    const enrollmentType = hasCertificate ? ModalType.CERTIFICATE_RENEWAL : ModalType.ENROLL;
-    await this.startEnrollment(enrollmentType);
-  }
-
-  public async attemptEnrollment(): Promise<void> {
-    const hasCertificate = await hasActiveCertificate();
-    if (hasCertificate) {
-      // If the client already has a certificate, we don't need to start the enrollment
-      return;
-    }
-    return this.startEnrollment(ModalType.ENROLL);
-  }
+  /**
+   * Will initiate the timer that will regularly prompt the user to enroll (or to renew the certificate if it is about to expire)
+   * @returns the delay under which the next enrollment/renewal modal will be prompted
+   */
+  public async startTimers() {
+    // We store the first time the user was prompted with the enrollment modal
+    const e2eActivatedAt = SnoozableTimerStore.get.e2eiActivatedAt() || Date.now();
+    SnoozableTimerStore.store.e2eiActivatedAt(e2eActivatedAt);
 
-  public async attemptRenewal(): Promise<void> {
+    const timerKey = 'enrollmentTimer';
     const identity = await getActiveWireIdentity();
+    const {firingDate, isSnoozable} = getEnrollmentTimer(identity, e2eActivatedAt, this.config.gracePeriodInMs);
 
-    if (!identity?.certificate) {
-      return;
-    }
-
-    const {timeRemainingMS, certificateCreationTime} = getCertificateDetails(identity.certificate);
-
-    if (!this.shouldRefresh(identity)) {
-      return;
-    }
-
-    // Check if an enrollment is already in progress
-    if (await this.coreE2EIService.isEnrollmentInProgress()) {
-      return this.enroll();
-    }
-
-    const renewalTimeMS = this.calculateRenewalTime(timeRemainingMS, historyTimeMS, this.config!.gracePeriodInMs);
-    const renewalPromptTime = new Date(certificateCreationTime + renewalTimeMS).getTime();
-    const currentTime = new Date().getTime();
+    const task = async () => this.processEnrollmentUponExpiry(isSnoozable);
 
-    // Check if it's time to renew the certificate
-    if (currentTime >= renewalPromptTime) {
-      await this.renewCertificate();
-      this.emit('identityUpdated', {enrollmentConfig: this.config!, identity});
+    if (TaskScheduler.hasActiveTask(timerKey)) {
+      TaskScheduler.continueTask({
+        key: timerKey,
+        task,
+      });
+    } else {
+      TaskScheduler.addTask({
+        key: timerKey,
+        task,
+        firingDate: firingDate,
+        persist: true,
+      });
     }
-  }
 
-  private shouldRefresh(identity: WireIdentity) {
-    const deviceIdentityStatus = identity.status;
-    switch (deviceIdentityStatus) {
-      case MLSStatuses.REVOKED:
-        return false;
-      default:
-        return true;
-    }
+    const fireIn = firingDate - Date.now();
+    this.logger.info(`Enrollment timer started {snoozable: ${isSnoozable}, fireAt: ${new Date(firingDate)}}`);
+    return fireIn;
```

Comment:
Can't by like that? :) You're not using fireIn in another places ;)

```suggestion
    this.logger.info(`Enrollment timer started {snoozable: ${isSnoozable}, fireAt: ${new Date(firingDate)}}`);
    return firingDate - Date.now();
```

Reply:
true that 👌